cmd/api/acrsdk.go

```diff
@@ -0,0 +1,208 @@
+}
+
+// LoginURL returns the FQDN for a registry
+func LoginURL(registryName string) string {
+	// TODO: if the registry is in another cloud (i.e. dogfood) a full FQDN for the registry should be specified.
+	if strings.Contains(registryName, ".") {
+		return registryName
+	}
+	return registryName + registryURL
+}
+
+// GetHostname return the hostname of a registry
+func GetHostname(loginURL string) string {
+	hostname := loginURL
+	if !strings.HasPrefix(loginURL, prefixHTTPS) {
+		hostname = prefixHTTPS + loginURL
+	}
+	return hostname
+}
+
+// AcrListTags list the tags of a repository with their attributes
+func AcrListTags(ctx context.Context,
+	loginURL string,
+	auth string,
+	repoName string,
+	orderBy string,
+	last string) (*acrapi.TagAttributeList, error) {
+	hostname := GetHostname(loginURL)
+	client := acrapi.NewWithBaseURI(hostname,
+		repoName,
+		"",
+		"",
+		"",
+		"",
+		auth,
+		orderBy,
+		"100",
+		last,
+		"")
+	if tags, e := client.AcrListTags(ctx); e == nil {
+		var listTagResult acrapi.TagAttributeList
+		switch tags.StatusCode {
+		case http.StatusOK:
+			if e = mapstructure.Decode(tags.Value, &listTagResult); e == nil {
+				return &listTagResult, nil
+			}
+			return nil, e
+
+		case http.StatusUnauthorized, http.StatusNotFound:
+			var apiError acrapi.Error
+			if e = mapstructure.Decode(tags.Value, &apiError); e == nil {
+				return nil, fmt.Errorf("%s %s", *(*apiError.Errors)[0].Code, *(*apiError.Errors)[0].Message)
+			}
+			return nil, errors.Wrap(e, "unable to decode error")
+
+		default:
+			return nil, fmt.Errorf("unexpected response code: %v", tags.StatusCode)
+		}
+	} else {
+		return nil, e
+	}
+}
+
+// AcrDeleteTag deletes the tag by reference.
+func AcrDeleteTag(ctx context.Context,
+	loginURL string,
+	auth string,
+	repoName string,
+	reference string) error {
+	hostname := GetHostname(loginURL)
+	client := acrapi.NewWithBaseURI(hostname,
+		repoName,
+		reference,
+		"",
+		"",
+		"",
+		auth,
+		"",
+		"",
+		"",
+		"")
+
+	if tag, e := client.AcrDeleteTag(ctx); e == nil {
+		switch tag.StatusCode {
+		case http.StatusAccepted:
+			return nil
+		case http.StatusBadRequest, http.StatusUnauthorized, http.StatusNotFound, http.StatusMethodNotAllowed:
+			var apiError acrapi.Error
+			if e = mapstructure.Decode(tag, &apiError); e == nil {
+				return fmt.Errorf("%s %s", *(*apiError.Errors)[0].Code, *(*apiError.Errors)[0].Message)
+			}
+			return errors.Wrap(e, "unable to decode error")
+
+		default:
+			return fmt.Errorf("unexpected response code: %v", tag.StatusCode)
+		}
+	} else {
+		return e
+	}
+}
+
+// AcrListManifests list all the manifest in a repository with their attributes.
+func AcrListManifests(ctx context.Context,
+	loginURL string,
+	auth string,
+	repoName string,
+	orderBy string,
+	last string) (*acrapi.ManifestAttributeList, error) {
+	hostname := GetHostname(loginURL)
+	client := acrapi.NewWithBaseURI(hostname,
+		repoName,
+		"",
+		"",
+		"",
+		"",
+		auth,
+		orderBy,
+		"100",
```

Comment:
Why `100`?

Reply:
I left it with 100 because it is the default number of tags that you can get using the list tags method.

Reply:
are we saying that only 100 tags are deleted each time?

Reply:
No, in the PurgeTags function there is a loop that deletes at most 100 tags on every iteration, it loops until it has finished scanning all the existent tags.